maestrowf/datastructures/core/executiongraph.py

```diff
@@ -48,6 +48,7 @@ def __init__(self, workspace, step, **kwargs):
         """
         self.workspace = Variable("WORKSPACE", workspace)
         step.run["cmd"] = self.workspace.substitute(step.run["cmd"])
+        step.run["cmd"] = self.workspace.substitute(step.run["restart"])
```

Comment:
Shouldn't this be `step.run["restart"] = self.workspace.substitute(step.run["restart"])` ?

Reply:
Confirmed: this bugfix only works completely with this change I've suggested.

Reply:
Good catch, that's what I get for coding on a plane while internet is cutting in and out. xD

I just pushed a change.